test/unit/task2/repository/userRepository.test.js

```diff
@@ -0,0 +1,147 @@
+const dataBase = require('./../../../../src/task2/repository/dataBase');
+
+const USER_ID = 'testid'
+const USER_LOGIN = 'login'
+const USER_PASSWORD = 'password'
+const USER_AGE = 'age'
+const IS_DELETED = false
+const user = {
+    id: USER_ID,
+    login: USER_LOGIN,
+    password: USER_PASSWORD,
+    age: USER_AGE,
+    isDeleted: IS_DELETED
+}
+
+beforeAll(() => {
+    jest.spyOn(dataBase, 'set')
+    jest.spyOn(dataBase, 'get')
+    jest.spyOn(dataBase, 'values')
+})
+
+describe('Create user', () => {
+    test('should return newly created user', async () => {
+        const result = await userRepository.saveUser(user)
+
+        expect(result).toEqual(user)
+    })
+
+    test('should pass expected arguments to set method of dataBase', async () => {
+        await userRepository.saveUser(user)
+
+        expect(dataBase.set).toHaveBeenCalledWith(USER_ID, user)
+    })
+})
+
+describe('Get user', () => {
+    let userCopy = {...user}
+
+    beforeAll(() => {
+        userCopy.password = '*****'
+        dataBase.set(USER_ID, user)
+    })
+
+    test('should return copy of existing user', async () => {
+        const result = await userRepository.getUser(USER_ID)
+
+        expect(result === user).toBe(false)
+        expect(result).toEqual(userCopy)
+    })
+
+    test('should return undefined when user is absent', async () => {
+        dataBase.delete(USER_ID)
+
+        const result = await userRepository.getUser(USER_ID)
+
+        expect(result).toEqual(undefined)
+    })
+
+    test('should pass expected argument to get method of dataBase', async () => {
+        await userRepository.saveUser(user)
+
+        expect(dataBase.get).toHaveBeenCalledWith(USER_ID)
+    })
+})
+
+describe('Delete user', () => {
+    let userCopy = {...user}
+
+    beforeAll(() => {
+        userCopy.isDeleted = true
+        dataBase.set(USER_ID, user)
+    })
+
+    test('should return copy of deleted user', async () => {
+        const result = await userRepository.deleteUser(user)
+
+        expect(result === user).toBe(false)
+        expect(result).toEqual(userCopy)
+    })
+
+    test('should pass expected arguments to set method of dataBase', async () => {
+        await userRepository.deleteUser(user)
+
+        expect(dataBase.set).toHaveBeenCalledWith(USER_ID, userCopy)
+    })
+})
+
+describe('Get auto suggest users', () => {
+    let secondUser
+    let userCopy
+    let secondUserCopy
+
+    beforeEach(() => {
+        secondUser = {
+            id: 'another' + USER_ID,
+            login: 'another' + USER_LOGIN,
+            password: USER_PASSWORD,
+            age: USER_AGE,
+            isDeleted: IS_DELETED
+        }
+        userCopy = {...user}
+        secondUserCopy = {...secondUser}
+        userCopy.password = '******'
+        secondUserCopy.password = '******'
+        dataBase.set(user.id, user)
+        dataBase.set(secondUser.id, secondUser)
+    })
+
+    afterEach(() => {
+        jest.clearAllMocks()
```

Comment:
dataBase is not mocked by jest, so jest.clearAllMocks() doesn't clear database, does it?

Reply:
it is needed to count 'values' calls for spy